aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -286,37 +337,260 @@ def setup(self):
+
+        if Aircraft.Engine.FIXED_RPM in self.aviary_inputs:
+            fixed_rpm = self.aviary_inputs.get_val(
+                Aircraft.Engine.FIXED_RPM, units='rpm'
+            )
 
-        outputs = ['*']
-
-        if Dynamic.Mission.THRUST in [
-            output_dict[key]['prom_name'] for key in output_dict
-        ]:
-            outputs.append((Dynamic.Mission.THRUST, 'turboshaft_thrust'))
+            if Dynamic.Mission.RPM in shp_output_list:
+                if self.aviary_inputs.get_val(Settings.VERBOSITY) >= Verbosity.BRIEF:
+                    warnings.warn(
+                        'Overriding RPM value outputted by EngineModel'
+                        f'{shp_model.name} with fixed RPM of {fixed_rpm}'
+                    )
 
-        if Dynamic.Mission.THRUST_MAX in [
-            output_dict[key]['prom_name'] for key in output_dict
-        ]:
-            outputs.append((Dynamic.Mission.THRUST_MAX, 'turboshaft_thrust_max'))
+                shp_outputs.append(
+                    (Dynamic.Mission.RPM, 'AUTO_OVERRIDE:' + Dynamic.Mission.RPM)
+                )
+                shp_output_list.remove(Dynamic.Mission.RPM)
+
+            fixed_rpm_nn = np.ones(self.num_nodes) * fixed_rpm
+
+            rpm_ivc.add_output(Dynamic.Mission.RPM, fixed_rpm_nn, units='rpm')
+            if has_gearbox:
+                self.promotes('fixed_rpm_source', [(Dynamic.Mission.RPM, 'fixed_rpm')])
+                gearbox_inputs.append((Dynamic.Mission.RPM + '_in', 'fixed_rpm'))
+                gearbox_input_list.remove(Dynamic.Mission.RPM + '_in')
+            else:
+                self.promotes('fixed_rpm_source', ['*'])
+        else:
+            rpm_ivc.add_output('AUTO_OVERRIDE:' + Dynamic.Mission.RPM, 1.0, units='rpm')
+            if has_gearbox:
+                if Dynamic.Mission.RPM in shp_output_list:
+                    shp_outputs.append(
+                        (Dynamic.Mission.RPM, Dynamic.Mission.RPM + '_gearbox')
+                    )
+                    shp_output_list.remove(Dynamic.Mission.RPM)
+                gearbox_inputs.append(
+                    (Dynamic.Mission.RPM + '_in', Dynamic.Mission.RPM + '_gearbox')
+                )
+                gearbox_input_list.remove(Dynamic.Mission.RPM + '_in')
+
+        # All other shp model outputs that don't interact with gearbox will be promoted
+        for var in shp_output_list:
+            shp_outputs.append(var)
+
+        #############################
+        # GEARBOX MODEL CONNECTIONS #
+        #############################
+        if has_gearbox:
+            # Promote all inputs which don't come from shp model (those got connected),
+            #   don't promote ones in skip list
+            for var in gearbox_input_list.copy():
+                if var not in skipped_inputs:
+                    gearbox_inputs.append(var)
+                # DO NOT promote inputs in skip list - always skip
+                gearbox_input_list.remove(var)
```

Comment:
should this be in an `else`? it is currently applied to every variable in gearbox_input_list

Reply:
I was taking a shortcut here, to be more verbose what I'm actually doing is this:
```
if var not in skipped_inputs:
    gearbox_inputs.append(var)
    gearbox_input_list.remove(var)
else:
    gearbox_input_list.remove(var)
```
 but figured that was redundant. The `gearbox_input_list` is tracking all the inputs I have yet to handle, either through promotion or connection. I end up handling every variable in the list here (either marking for promotion, or making sure it is entirely skipped), so deleting everything from `gearbox_input_list` was intended.